### ISSUE TITLE
feature request: public login page

### ISSUE THREAD

**Issue by michaelkarlcoleman** [OSC/ood-dashboard#427]
Somewhat similar to the logout page, it would be nice to have a login page of some sort that is shown to users prior to authenticating.  Currently, at least with Basic Auth, they're given a blank page and a dialog box to type their username/password into.  It would be nice to be able to show a page describing what they're logging into, public announcements, and some basic security advice (beware of public terminals, use "private browsing" mode if possible, etc.)

**Comment by ericfranz** [comment]
I'm not sure how we would implement this with Basic Auth. I don't know of a way to have a custom login form for a basic auth protected location. One approach would be to add an unauthenticated "start page" that / serves, and then /pun/* to be authenticated via basic auth like it is right now, instead of / redirecting to /pun/sys/dashboard.

This is the redirect:

https://github.com/OSC/ondemand/blob/799e7b39bfac5bc89f81306ee673dfdc3c0c7ec3/ood-portal-generator/templates/ood-portal.conf.erb#L221-L228

Notice the `root_uri` is a configuration option in ood-portal-generator. You enable the public root https://github.com/OSC/ondemand/blob/799e7b39bfac5bc89f81306ee673dfdc3c0c7ec3/ood-portal-generator/templates/ood-portal.conf.erb#L110-L122, drop an start.html page in there, and set the root_uri to be the URI to this static HTML page.

Then, unauthenticated users could access this, 

Perhaps on this page you could do an ajax request to the dashboard app at /pun/sys/dashboard and if the request fails you know that the user is not authenticated. That way, the JavaScript on the page could force a redirect to /pun/sys/dashboard, so when the user goes to your.ondemand.site.edu they end up at the dashboard instead of the start page.

Of course omitting the JavaScript makes things easier. Just when going directly to your.ondemand.site.edu they will see the start page.

The start page of course would have a link to the dashboard app, which is protected with basic auth. Is a start page like this a good idea out of the box? Maybe. I don't know.

**Comment by ericfranz** [comment]
Seems like it would be good if we continue to use Basic Auth as the default.

**Comment by michaelkarlcoleman** [comment]
> One approach would be to add an unauthenticated "start page" that / serves, and then /pun/* to be authenticated via basic auth like it is right now

Yes, this is what I had in mind.  And it sounds like it's not that hard to switch to this--I was missing that.

In our environment at least, it will be very useful to have a public "front door", where we can explain what the site is, how users get an account, and perhaps have some critical news (e.g., auth is failing due to an outage).  In addition, it provides an additional modicum of assurance that they're actually typing their password into the site they think they are (as opposed to having fat-fingered something).

Thanks!
